osbs/api.py

```diff
@@ -713,6 +713,47 @@ def get_build_logs(self, build_id, follow=False, build_json=None, wait_if_missin
 
         return logs
 
+    def _parse_build_log_entry(self, entry):
+        items = entry.split()
+        if len(items) < 4:
+            logger.warning("This is not a valid build log entry: %s", entry)
+            return (None, entry)
+        date = items[0]
+        time = items[1]
+        platform = items[2]
+        plen = sum(len(field) + 1  # include trailing space
+                   for field in [date, time, platform])
+        if not platform.startswith("platform:"):
+            logger.warning("Line logged without using the appropriate LoggerAdapter")
+            platform = None
+        else:
+            platform = platform.split(":")[1]
+        if platform == "-":
+            platform = None
+        line = entry[plen:]
+        return (platform, line)
+
+    @osbsapi
+    def get_orchestrator_build_logs(self, build_id, follow=False, wait_if_missing=False):
+        """
+        provide logs from orchestrator build
+
+        :param build_id: str
+        :param follow: bool, fetch logs as they come?
+        :param wait_if_missing: bool, if build doesn't exist, wait
+        :return: tuple with fields 'platform' and 'line'
+        """
+        logs = self.get_build_logs(build_id=build_id, follow=follow,
+                                   wait_if_missing=wait_if_missing, decode=True)
+
+        if not isinstance(logs, GeneratorType):
+            for entry in logs.split("\n"):
+                yield self._parse_build_log_entry(entry)
+        else:
+            for entries in logs:
+                for entry in entries.split("\n"):
```

Comment:
I think the generator function will always return "a log entry", in as much as it calls `iterlines()` on the streamed `requests.Response` object and yields what it parses from each JSON object. Given that, I don't think we need to then iterate over each line within that as it will only be a single line anyway.